underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/EntityGroupFilterBuilderBase.java

```diff
@@ -0,0 +1,364 @@
+                      notPrimaryEntity, new ArrayList<>(List.of(idSubFilter)));
+                }
+                EntityGroupFilterUtils.buildAllModifierFilters(
+                    underlay,
+                    List.of(notPrimaryEntity),
+                    criteriaSelector,
+                    valueData,
+                    modifiersSelectionData,
+                    filtersForSingleEntityGroup);
+                break;
+              default:
+                throw new SystemException(
+                    "Unsupported entity group type: " + entityGroup.getType());
+            }
+
+            List<EntityOutput> entityOutputsForSingleEntityGroup =
+                EntityGroupFilterUtils.mergeFiltersForDataFeature(
+                    filtersForSingleEntityGroup, BooleanAndOrFilter.LogicalOperator.AND);
+            entityOutputsForSingleEntityGroup.forEach(
+                entityOutput -> {
+                  List<EntityFilter> filters =
+                      filtersPerEntity.getOrDefault(entityOutput.getEntity(), new ArrayList<>());
+                  if (entityOutput.hasDataFeatureFilter()) {
+                    filters.add(entityOutput.getDataFeatureFilter());
+                  } else {
+                    filters = new ArrayList<>();
+                  }
+                  filtersPerEntity.put(entityOutput.getEntity(), filters);
+                });
+          });
+
+      // If there are multiple filters for a single entity, OR them together.
+      return EntityGroupFilterUtils.mergeFiltersForDataFeature(
+          filtersPerEntity, BooleanAndOrFilter.LogicalOperator.OR);
+    }
+  }
+
+  private Map<EntityGroup, List<Literal>> selectedIdsPerEntityGroup(
+      Underlay underlay, String serializedSelectionData) {
+    Map<EntityGroup, List<Literal>> selectedIdsPerEntityGroup = new HashMap<>();
+    Map<Literal, String> selectedIdsAndEntityGroups =
+        selectedIdsAndEntityGroups(serializedSelectionData);
+    selectedIdsAndEntityGroups.forEach(
+        (key, entityGroupId) -> {
+          EntityGroup entityGroup = underlay.getEntityGroup(entityGroupId);
+          List<Literal> selectedIds =
+              selectedIdsPerEntityGroup.containsKey(entityGroup)
+                  ? selectedIdsPerEntityGroup.get(entityGroup)
+                  : new ArrayList<>();
+          selectedIds.add(key);
+          selectedIdsPerEntityGroup.put(entityGroup, selectedIds);
+        });
+
+    selectedIdsPerEntityGroup.forEach(
+        (entityGroup, selectedIds) -> {
+          Collections.sort(selectedIds);
+        });
+    return selectedIdsPerEntityGroup;
+  }
+
+  private List<EntityGroup> selectedEntityGroups(
+      Underlay underlay, Map<EntityGroup, List<Literal>> selectedIdsPerEntityGroup) {
+    List<EntityGroup> selectedEntityGroups;
+    if (!selectedIdsPerEntityGroup.isEmpty()) {
+      selectedEntityGroups = new ArrayList<>(selectedIdsPerEntityGroup.keySet());
+    } else {
+      selectedEntityGroups =
```

Comment:
That's actually not the case because `buildForCohort` and `buildForDataFeature` early exit for criteria with no selection or modifiers. In the modifiers only case this is a change as expected.

That being said, I think changing the behavior is actually correct. Adding an empty condition filter should filter on all people who have a condition rather than doing nothing. I'll probably follow up to remove the early exit but I didn't want to put anything else into this PR if I didn't need to.